src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -401,15 +442,43 @@ void ClusterQueueHelper::afterPartitionPrimaryAssignment(
     }
 }
 
-void ClusterQueueHelper::assignQueue(const QueueContextSp& queueContext)
+bool ClusterQueueHelper::assignQueueIfNeeded(
+    const QueueContextSp& queueContext_sp)
 {
     // executed by the cluster *DISPATCHER* thread
 
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(
         d_cluster_p->dispatcher()->inDispatcherThread(d_cluster_p));
+
+    BSLS_ASSERT_SAFE(queueContext_sp);
+
+    bool result = isQueueAssigned(*queueContext_sp);
+
+    if (!result) {
+        // Queue is not assigned to a partition; get it assigned.  If
+        // self is leader, it will assign it locally, if not it will
+        // send a request to the leader, etc.
+
+        result = assignQueue(queueContext_sp);
+    }
+
+    return result;
```

Comment:
Possible alternative
```suggestion
    if (isQueueAssigned(*queueContext_sp)) {
        // Already assigned, nothing to do
        return true;  // RETURN
    }

    // Queue is not assigned to a partition; get it assigned.  If
    // self is leader, it will assign it locally, if not it will
    // send a request to the leader, etc.
    return assignQueue(queueContext_sp);
```